ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -825,6 +859,35 @@ impl<'a> X64CodeGen<'a> {
         self.store_new_local(inst_idx, WR0);
     }
 
+    fn cg_trunc(&mut self, inst_idx: InstIdx, i: &jit_ir::TruncInst) {
+        let from_val = i.val();
+        let from_type = self.m.type_(from_val.ty_idx(self.m));
+        let from_size = from_type.byte_size().unwrap();
+
+        let to_type = self.m.type_(i.dest_ty_idx());
+        let to_size = to_type.byte_size().unwrap();
+
+        debug_assert!(matches!(to_type, jit_ir::Ty::Integer(_)));
+        debug_assert!(
+            matches!(from_type, jit_ir::Ty::Integer(_)) || matches!(from_type, jit_ir::Ty::Ptr)
+        );
+        // You can only truncate a bigger integer to a smaller integer.
+        debug_assert!(from_size > to_size);
+
+        // FIXME: assumes the input and output fit in a register.
+        self.load_operand(WR0, &from_val);
+        debug_assert!(to_size <= REG64_SIZE);
+
+        // FIXME: There's no instruction on x86_64 to mov from a bigger register into a smaller
```

Comment:
I *think* that if you trunc to a byte boundary, you can just do nothing and leave the top bytes undefined. The IR type should dictate how many bytes should be read/written when the value is later used?

If you trunc to a non-byte value, you will be forced to zero some bits though. Right?

Reply:
That matches my understanding.